CannibalisticZombies/Assets/Scripts/Inventory.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class Inventory 
+{
+    private List<InventorySlot> inventory;
+    private int maxWeight;
+
+    public Inventory() 
+    {
+        inventory = new List<InventorySlot>();
+    }
+
+    //-/////////////////////////////////////////////////////////////////////
+    /// adds amount of new item to existing InventorySlot with said item
+    /// if maxWeight allows
+    public void AddItem(Item newItem, int amount = 1)
+    {
+        // return if adding goes overweight
+        if (GetCurrentWeight()+newItem.weight * amount > maxWeight) { return; }
+        foreach (InventorySlot slot in inventory) 
```

Comment:
There are methods to handle searching for an item!
In a comment above, I recommended using a dictionary! You can use methods like` ContainsKey(keyName)` to check if the item is already in the inventory, and then add to it that way.